src/components/Home/WalletCard.tsx

```diff
@@ -0,0 +1,82 @@
+import { Card } from '@/components/0_Bruddle'
+import Image from 'next/image'
+import { motion } from 'framer-motion'
+import classNames from 'classnames'
+import Link from 'next/link'
+import { formatUnits } from 'viem'
+import PeanutWalletIcon from '@/assets/icons/peanut-wallet.png'
+import Icon from '@/components/Global/Icon'
+import { shortenAddressLong } from '@/utils'
+import { IWallet } from '@/interfaces'
+
+interface WalletCardProps {
+    type: 'wallet' | 'add'
+    wallet?: IWallet
+    username?: string
+    selected?: boolean
+    onClick?: () => void
+}
+
+export function WalletCard({ type, wallet, username, selected = false, onClick }: WalletCardProps) {
+    if (type === 'add') {
+        return (
+            <motion.div className="h-full">
+                <Card
+                    className="h-full min-w-[300px] rounded-md text-black hover:cursor-pointer"
+                    shadowSize="6"
+                    onClick={onClick}
+                >
+                    <Link href="/setup" className="h-full">
+                        <Card.Content className="flex h-full flex-col items-center justify-center gap-6 p-6">
+                            <p className="text-center text-xl font-bold">Add your own ETH wallet</p>
+                            <div className="flex flex-row items-center gap-3">
+                                <Icon name="plus-circle" className="h-6 w-6" />
+                                <span className="text-base">Add BYOW wallet</span>
+                            </div>
+                        </Card.Content>
+                    </Link>
+                </Card>
+            </motion.div>
+        )
+    }
+
+    if (!wallet) return null
```

Comment:
Q: whats the point of this? why does wallet have to be optional? Wouldnt it be better to force it?

Reply:
I agree, modifying